src/coreclr/gc/gc.cpp

```diff
@@ -18018,11 +18042,26 @@ bool gc_heap::update_alloc_info (int gen_number, size_t allocated_size, size_t*
 
     size_t& etw_allocated = etw_allocation_running_amount[oh_index];
     etw_allocated += allocated_size;
-    if (etw_allocated > etw_allocation_tick)
+
+    size_t& etw_threshold = etw_allocation_running_threshold[oh_index];
+    if (etw_allocated > etw_threshold)
     {
         *etw_allocation_amount = etw_allocated;
         exceeded_p = true;
         etw_allocated = 0;
+
+// avoid computing if not needed
+#ifdef FEATURE_EVENT_TRACE
+  #ifdef FEATURE_NATIVEAOT
+    if (EVENT_ENABLED(GCAllocationTick_V1))
+  #else
+    if (EVENT_ENABLED(GCAllocationTick_V4))
+  #endif
+    {
+        // compute the next threshold based on a Poisson process with a etw_allocation_tick_mean average
```

Comment:
many of us aren't greatly familiar with statistics so making this explanation not so vague would be helpful. instead of saying "based on a Possion process" it'd be much more helpful to start with something like "we are treating this as a Possion process because each sample we take has no influence on any other sample. the samples are exponentially distributed in a Possion process, meaning that the possibility of the next sample happening is calculated by `(1 - e^(-lambda*x))`. and then explain what lambda and x would be in this particular context so the readers know how the formula you are using came to be. 

also `-ln (1 - uniformly_random_number_between_0_and_1)`, is the same as `-ln (uniformly_random_number_between_0_and_1)`. so I don't think you need the `1 - ` part. 

can you please show the results of running this on some workloads where this is much better compared to the current implementation? also have you tried with just a uniformly random distribution instead of an exponential distribution?

Reply:
> many of us aren't greatly familiar with statistics so making this explanation not so vague would be helpful. instead of saying "based on a Possion process" it'd be much more helpful to start with something like "we are treating this as a Possion process because each sample we take has no influence on any other sample. the samples are exponentially distributed in a Possion process, meaning that the possibility of the next sample happening is calculated by `(1 - e^(-lambda*x))`. and then explain what lambda and x would be in this particular context so the readers know how the formula you are using came to be.

I updated the description accordingly with also additional information about the upscaling formula

Reply:
> can you please show the results of running this on some workloads where this is much better compared to the current implementation? also have you tried with just a uniformly random distribution instead of an exponential distribution?

I'm currently simulating the results based on a web application for which I'm recording ALL allocations using ICorProfilerCallback::ObjectAllocated() and check against the sampled then upscaled sizes. The variance of the results shows almost random results for fixed threshold, much better for variable threshold as in the first commit and a little better if sampling could happen within allocation context.
Since the recorder is available in the Datadog profiler only, it will be complicated to generate the corresponding .balloc files (i.e. list of allocations - type+size) used by the simulation to show result on any application. BTW, is there any sample application that you would like to see used as example?

Reply:
> also -ln (1 - uniformly_random_number_between_0_and_1), is the same as -ln (uniformly_random_number_between_0_and_1). so I don't think you need the 1 -  part.

This sticks to the mathematical way to derive the formula. Since the result should be the same, I would recommend to keep it as it is.